src/lib.rs

```diff
@@ -0,0 +1,198 @@
+extern crate regex;
+use regex::Regex;
+use std::error;
+use std::fmt;
+
+
+pub struct Siso {
+    route: Route,
+}
+
+impl Siso {
+    pub fn register(&mut self, path_nodes : &[PathNode], value: RouteValue) -> Result<bool, SisoError> {
+        if 0 == path_nodes.len() {
+            panic!("Need a least one node for registering a value...");
+        }
+        return self.route.register(path_nodes, value);
```

Comment:
No need for return here, just `self.route.register(path_nodes, value)` (*without* `;`)